docs/BacwardCompatibility.md

```diff
@@ -0,0 +1,133 @@
+[Kubernetes](https://kubernetes.io/docs/reference/using-api/#api-versioning).
+- Alpha
+    - The version names contain alpha (for example, v1alpha1).
+    - The software is not stable may contain bugs.
+    - The support for a feature may be dropped at any time without notice.
+    - The API may change in incompatible ways in a later software release without notice.
+- Beta
+    - The version names contain beta (for example, v2beta3).
+    - The software is well tested. Enabling a feature is considered safe.
+    - The support for a feature will not be dropped, though the details may change.
+    - The schema and/or semantics of objects may change in incompatible ways in a subsequent beta or stable release.
+      When this happens, migration instructions are provided. Schema changes may require deleting, editing, and re-creating
+      API objects. The editing process may not be straightforward. The migration may require downtime for applications that
+      rely on the feature.
+- Stable
+    - The version name is vX where X is an integer.
+    - The stable versions of features appear in released software for many subsequent versions.
+### Relations between CRD changes and Fybrik releases
+Based on the above definitions, the relation between CRD versions and Fybrik versions is unidirectional. Which means,
+CRD API changes requires at least minor Fybrik release, on the other hand not each Fybrik release requires a new CRD
+version.
+
+### Upgrade Plan
+The CustomResourceDefinition API provides a workflow for introducing and upgrading to new versions of a 
+CustomResourceDefinition. For more information about 
+[serving multiple versions](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#serving-multiple-versions-of-a-crd) of your CustomResourceDefinition 
+and migrating your objects from one version to another see 
+[Versions in CustomResourceDefinitions](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/)
+
+The first step of adding a new version is to "pick a conversion strategy. Since custom resource objects need to be able 
+to be served at both versions, that means they will sometimes be served at a different version than their storage 
+version. In order for this to be possible, the custom resource objects must sometimes be converted between the version 
+they are stored at and the version they are served at. If the conversion involves schema changes and requires custom 
+logic, a conversion **Webhook** should be used. If there are no schema changes, the default **None** conversion strategy may 
+be used and only the apiVersion field will be modified when serving different versions."
+
+Due to some Fybrik customers' limitations, we cannot use Webhooks in Fybrik deployments. Therefore, below we investigate 
+possible CRD changes without changing its schema.
```

Comment:
Migration tools can be provided that are not webhook based if necessary.  Not sure we can assume we will never change the schema in such a way that migration is required.  We can state that we should aim to do so, but we can't guarantee that.